pkg/kubectl/resource_filter.go

```diff
@@ -39,18 +37,9 @@ func NewResourceFilter() Filters {
 
 // filterPods returns true if a pod should be skipped.
 // If show-all is true, the pod will be never be skipped (return false);
-// otherwise, skip terminated pod.
-func filterPods(obj runtime.Object, options printers.PrintOptions) bool {
-	if options.ShowAll {
-		return false
-	}
-
-	switch p := obj.(type) {
-	case *v1.Pod:
-		return p.Status.Phase == v1.PodSucceeded || p.Status.Phase == v1.PodFailed
-	case *api.Pod:
-		return p.Status.Phase == api.PodSucceeded || p.Status.Phase == api.PodFailed
-	}
+// show-all is deprecated in v1.10 and inert in v1.11.
+// TODO: It will be removed in a future release.
+func filterPods(runtime.Object, printers.PrintOptions) bool {
```

Comment:
why is this still here?

Reply:
I thought it should be removed in the future. Should I remove it in this PR? @deads2k

Reply:
The inert bit is about the flag which needs to continue to exist so someone specifying it doesn't get a failure.  We don't need to keep cruft in the codebase though.

Reply:
@deads2k Rebased.